Description: fix reading it->extra on big endian when bytesNeeded == 1
Origin: upstream, https://codereview.qt-project.org/258957
Last-Update: 2019-04-19

--- a/src/3rdparty/tinycbor/src/cborparser.c
+++ b/src/3rdparty/tinycbor/src/cborparser.c
@@ -203,10 +203,13 @@ static CborError preparse_value(CborValu
         it->extra = 0;
 
         /* read up to 16 bits into it->extra */
-        if (bytesNeeded <= 2) {
+        if (bytesNeeded == 1) {
+            uint8_t extra;
+            read_bytes_unchecked(it, &extra, 1, bytesNeeded);
+            it->extra = extra;
+        } else if (bytesNeeded == 2) {
             read_bytes_unchecked(it, &it->extra, 1, bytesNeeded);
-            if (bytesNeeded == 2)
-                it->extra = cbor_ntohs(it->extra);
+            it->extra = cbor_ntohs(it->extra);
         } else {
             cbor_static_assert(CborIteratorFlag_IntegerValueTooLarge == (Value32Bit & 3));
             cbor_static_assert((CborIteratorFlag_IntegerValueIs64Bit |
